speechbrain/lobes/models/huggingface_wav2vec.py

```diff
@@ -211,31 +213,62 @@ def _load_sb_pretrained_w2v2_parameters(self, path):
                 + "is useless for wav2vec 2.0 finetuning."
             )
 
-    def _check_model_source(self, path):
+    def _check_model_source(self, path, save_path):
         """Checks if the pretrained model has been trained with SpeechBrain and
         is hosted locally or on a HuggingFace hub.
+        Called as static function in HuggingFaceTransformer._from_pretrained.
+        Arguments
+        ---------
+        path : str
+            Used as "source"; local path or HuggingFace hub name: e.g "facebook/wav2vec2-large-lv60"
+        save_path : str
+            norm_output (dir) of the downloaded model.
+        Returns
+        -------
+        is_sb : bool
+            Whether/not the model is deserializable w/ SpeechBrain or not (then, model conversion is needed).
+        checkpoint_filename : str
+            as of HuggingFace documentation: file name relative to the repo root (guaranteed to be here).
         """
         checkpoint_filename = ""
         source = pathlib.Path(path)
         is_local = True
-        is_sb = True
 
         # If path is a huggingface hub.
         if not source.exists():
             is_local = False
 
+        # Check if source is downloaded already
+        sink = pathlib.Path(
```

Comment:
What is this for exactly? Isn't HuggingFace already not re-downloading when it's there?

Reply:
I think the issue was when being offline and the TRANSFORMERS=OFFLINE & DATASETS=OFFLINE (or similar) env params were not set. Some while ago; kept reappearing for some reason (too long ago).

Reply:
These lines avoid to trigger HF scripts when pretrained model data is locally present already. At the time, I did not know about `TRANSFORMERS=OFFLINE & DATASETS=OFFLINE` env flags. There are two issues here:
* if your server isn't connected, you get stuck (unless you know about these flags)
* if your server is connected & data is fetched, HF fetch will send a msg to get latest meta information (which can be a feature). Not only HF will have access to that msg.

> Isn't HuggingFace already not re-downloading when it's there?

They'll try to fetch meta data of the modoel card, regardless. This snippet prevents this. Btw, this edit corresponds to freezing dependencies, with pros/cons of that. The additional benefit is less inet traffic about what you do.